dds/DCPS/DataWriterImpl_T.h

```diff
@@ -341,82 +341,40 @@ DataWriterImpl_T
 
   DDS::ReturnCode_t setup_serialization()
   {
-    /**
-     * TODO(iguessthislldo): Update when we can determine what encoding we need
-     * to use from from a data representation QoS.
-     */
-    encoding_mode_ = EncodingMode(
-      cdr_encapsulation() ?
-        Encoding::KIND_XCDR1 : Encoding::KIND_UNALIGNED_CDR,
-      swap_bytes());
-#if 0
-    /*
-     * We need to get the encoding kinds for encapsulated and non-encapsulated
-     * encoding modes. We do this by picking the first supported representation
-     * that is supported by the mode, which is sorta what the XTypes spec says
-     * to do, except it only assumes one mode. One mode (but not both) may be
-     * left set to KIND_UNKNOWN, disabling it and preventing it from matching
-     * with readers just available through that mode.
-     */
-    Encoding::Kind encap_kind = Encoding::KIND_UNKNOWN;
-    Encoding::Kind nonencap_kind = Encoding::KIND_UNKNOWN;
-    DDS::DataRepresentationIdSeq repIds =
-      get_effective_data_rep_qos(qos_.representation.value);
-    for (CORBA::ULong i = 0; i < repIds.length(); ++i) {
-      const DDS::DataRepresentationId_t repr = repIds[i];
-      const Encoding::Kind kind =
-        repr_ext_to_encoding_kind(repr, MarshalTraitsType::extensibility());
-      switch (kind) {
-      case Encoding::KIND_CDR_UNALIGNED:
-        if (nonencap_kind == Encoding::KIND_UNKNOWN) {
-          nonencap_kind = kind;
+    DDS::DataRepresentationIdSeq repIds = get_effective_data_rep_qos(qos_.representation.value);
+    bool success = false;
+    Encoding::Kind encoding_kind;
+    if (cdr_encapsulation()) {
+      for (CORBA::ULong i = 0; i < repIds.length(); ++i) {
+        // TODO(sonndinh): Do we need to return failure when encountering an invalid representation
```

Comment:
Okay. May the data representations in `qos_.representation.value` contains some invalid values different than XCDR, XCDR2, XML, and UNALIGNED?

Reply:
It shouldn't, but I don't think it hurts to assume it's possible.

Reply:
In that case, should it stop and return error, or just skip the invalid data representations and keep going?

Reply:
It looks like my code returned an error, but I don't think it matters much as long as it can differentiate unsupported valid representations from completely invalid representations.